### ISSUE TITLE
Feature request: Show all graphs from host

### ISSUE THREAD

**Issue by p0nt** [Mikesch-mp/icingaweb2-module-grafana#30]
So currently you can see the graphs per service when you click on the service, but it would be very great if there would be a button or drop down menu to choose if you would like to show all graphs from the specific host underneath each other.

Personally I have hacked this into icingaweb2 with custom php files which is very very ugly, but it looks like this:

http://shot.xsnews.nl/stefan/e605b8a9e04d11dd.png

And when you click on All Graphs:

http://shot.xsnews.nl/stefan/dc6dc121c5ed47ed.png

It simply just loops trough all the host its services, and show them one by one.

I suggest some kind of show/hide div which will load after clicking it, because loading these graphs take a bit of time it will otherwise make the service information come very late.

something like this:

http://shot.xsnews.nl/stefan/3684d8b7c842052f.png

http://shot.xsnews.nl/stefan/3605bddd41e76e11.png

So basicaly what I would like is whenever you are on the host tab (not service tab), be able to click something and show every single graph from that specific host.

**Comment by Mikesch-mp** [comment]
I develpoed a dashboard for [Grafana ](https://grafana.com/dashboards/1572) a while ago, this + a shared Navigation link and you have a page that shows all graphs at once.

**Comment by p0nt** [comment]
Thank you, this is nice but not what I mean.

With this dashboard you will have all graphs with same settings, which means custom series overwrites or graphs that need to output data rate instead of normal number will not work.

Would be great if it will loop through the services, check if the service is available here:
http://icinga.xsnews.intern/icingaweb2/grafana/graph#!/icingaweb2/grafana/graph/new
and if so, show the graph.

**Comment by dnsmichi** [comment]
I don't think that this would fit into the host detail view. a) it shouldn't be loaded by default when opening the host detail view b) fetching multiple images from grafana is fairly slow. 

Imho such dashboards really should be rendered inside Grafana itself where one can zoom and selectively work with the graphs. Putting sort of such a dashboard into Icinga Web 2 reduces its flexibilty and feature parity with existing dashboards.

**Comment by p0nt** [comment]
Yes, thats why I suggested a seperate page or div that get loaded only when clicked. Also lazyloading the graphs would help the page for becoming usable straight away and whike you scroll down the graphs they automatically load without a problem.

**Comment by p0nt** [comment]
While you scroll down to have a look at the graphs*

**Comment by epinter** [comment]
I don't have all graphs in a single page on icinga (that would kill the UX of icinga user), but I have a link to a dashboard to all graphs. Today I have dedicated dashboards for all possible services, one 
dashboard for linux and one for windows servers. I'm using a feature to group hosts overriding panelId and dashboard names, based on hostgroups or hostnames, inside graphs definitions. And another feature to specify the dashboard the user will open when it clicks on the host graph (eg. hostalive check of linuxservers would point to dashboard named 'linux').

This is what I'm working on. It's not complete yet, so I'm doing fast-forward push to git...

https://github.com/epinter/icingaweb2-module-grafana/commit/600d43e0574b19bfa7160d263f41e5cc28f61712

https://github.com/epinter/icingaweb2-module-grafana/commit/85248f2aaef374f067049bba2e3f294575fc3469

https://github.com/epinter/icingaweb2-module-grafana/commit/19317fcfca38a01b2d3fe86bc61ea4dcaa8f9a01

**Comment by p0nt** [comment]
Sounds great, unfortunatly will have a look at it after the weekend.

Will keep you posted

**Comment by Mikesch-mp** [comment]
@epinter thats not something i would add because it makes it more complicated nad configuration can cause errors i dont want to suport. Soon i have 13 days free time and i will release 1.2 then with the feature to show all service graphs on one page of a host. And this will not need any configuration or special dashboards.

**Comment by epinter** [comment]
@Mikesch-mp, I have no problem about you accepting my changes or not. I'm trying to develop this module to fit my needs and publish here because other people may have the same problem as me. I'm trying to contribute.

Your idea sounds very nice, I think will help a lot of people that wants to view everying about a host. But personally I would never use a feature that shows me 15 graphs inside the host detail view of icinga (sorry if I'm misunderstanding you). Anyway, I hope you solve the performance issue that all graphs on a single page will introduce (at least when the grafana/influx server needs to handle thousands of services, which is my case).

**Comment by Mikesch-mp** [comment]
@epinter its only a link to a other page inside of icingaweb2

**Comment by epinter** [comment]
Much better then.

**Comment by p0nt** [comment]
@Mikesch-mp have any progress on this? I am able to maybe help you test or something, would love to see this feature (was trying to hack this into the module myself but failed for now)

**Comment by Mikesch-mp** [comment]
@p0nt only some very early tests on my dev system. At the moment iam very busy, but soon i have 2 weeks free time :) I cant givbe you an exact date now, but around end  of July i will have it ready for testing.

**Comment by p0nt** [comment]
sounds good thank you so much!

**Comment by p0nt** [comment]
great work with the all graphs per dashboard update! any info about the progress of this?

**Comment by jagdishrox** [comment]
![icingaproblemingraph](https://user-images.githubusercontent.com/28803608/32321210-66f53e30-bfe6-11e7-8577-40051d9aaa59.png)
########################################
i get this error.... i do all steps according to your link but found this error like above page..... 
plz suggest what will i do.....
******************************
io am new for icinga2 , and trying grafana module in icinga2 implementation..... i download your zip file and extract into /usr/share/icingaweb2/modules/grafana ..below image shows the all files inside grafana
![grfanafiles](https://user-images.githubusercontent.com/28803608/32321482-49804740-bfe7-11e7-9097-fca9c9ca2bb2.png)
***************************
plz let me know where i doing mistake.. plz help me......

**Comment by jagdishrox** [comment]
here is the configuration details i used 
![grafanaconfiguration](https://user-images.githubusercontent.com/28803608/32322538-d832690c-bfea-11e7-83a5-2faa105b8da2.png)

##########################

**Comment by dnsmichi** [comment]
Please join the community channels instead. Putting such unrelated questions into this feature request makes it hard to follow.

**Comment by p0nt** [comment]
any new information about this feature yet?

**Comment by Mikesch-mp** [comment]
See https://twitter.com/Mikeschova/status/964990487336759297

**Comment by p0nt** [comment]
awesome awesome!

**Comment by p0nt** [comment]
Just want to let you know that it works flaweless! It looks awesome.

The only side note is that it shows text for every single check the server has but not always that check have a graph.

**Comment by p0nt** [comment]
I found it, I just change this myself:

`+            if ( ! property_exists($object, 'service_description') && isset($object->service_description) ) {
+                echo "<h2>" . $object->service_description . "</h2>";
+            } else {
+                echo "<h2>" . $object->check_command . "</h2>";
+            }`

**Comment by Mikesch-mp** [comment]
Did you test with the latest master ?

**Comment by p0nt** [comment]
yeah, downloaded the new version today :) 

its great! works awesome

**Comment by Mikesch-mp** [comment]
Ahh got it, you forgot to disable perfdata on services that do not generate any perfdata?

**Comment by dnsmichi** [comment]
A bit hard to find through the host action, therefore I've created #125 

Looks nice indeed :)

<img width="1528" alt="screen shot 2018-03-04 at 19 53 33" src="https://user-images.githubusercontent.com/382049/36949273-54da690e-1fe6-11e8-8621-659426917df0.png">
<img width="593" alt="screen shot 2018-03-04 at 19 53 39" src="https://user-images.githubusercontent.com/382049/36949274-54f52c44-1fe6-11e8-94cc-19acd13e20a4.png">


Keep in mind though, that a host with 20+ graphs pulls a lot from Grafana. You'll recognize that with a small sized Vagrant VM, or low level VM/container in production. Grafana really needs power to serve things fast enough, even with the direct auth.anonymous access.
